controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -425,6 +476,41 @@ func (r *StorageClassRequestReconciler) reconcileCephBlockPool(storageProfile *v
 	return nil
 }
 
+func (r *StorageClassRequestReconciler) reconcileRadosNamespace(storageProfile *v1.StorageProfile) error {
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephRadosNamespace, func() error {
+		if err := r.own(r.cephRadosNamespace); err != nil {
+			return err
+		}
+
+		addLabel(r.cephRadosNamespace, controllers.StorageConsumerNameLabel, r.storageConsumer.Name)
+		addLabel(r.cephBlockPool, controllers.StorageProfileSpecLabel, storageProfile.GetSpecHash())
+
+		r.cephRadosNamespace.Spec = rookCephv1.CephBlockPoolRadosNamespaceSpec{
+			BlockPoolName: r.cephBlockPool.Name,
```

Comment:
The code is correct, but I was not able to locate where you search for a blockpool named with the profile name on the request (or an alternative profile annotation). This is important because that blockpool should be the one named here as the desired state.